src/grape/hooks/auth_field.clj

```diff
@@ -7,60 +7,40 @@
       (let [auth-strategy (:auth-strategy resource)
             auth-field? (= (:type auth-strategy) :field)
             doc-field (:doc-field auth-strategy)
+            additional-permission-doc-fields (or (:additional-permission-doc-fields auth-strategy) [])
             request-auth (:auth request)
             auth-value ((:auth-field auth-strategy) request-auth)]
         (if auth-field?
           (if (not request-auth)
             (throw (ex-info "Unauthorized" {:type :unauthorized}))
-            (on-request-auth auth-value doc-field arg))
+            (on-request-auth auth-value doc-field additional-permission-doc-fields arg))
           arg))
       arg)))
 
+(def has-rights-over [payload doc-field additional-permission-doc-fields auth-value]
+  (let [fields (concat [(get payload doc-field)] (map additional-permission-doc-fields #((get payload %))))]
+    (contains? fields auth-value))
+
 (def hooks
   {:pre-create-pre-validate
-   (wrap-auth :create (fn [auth-value doc-field payload]
-                        (update-in payload [doc-field]
-                                   (fn [existing-value]
-                                     (if-not existing-value
-                                       auth-value
-                                       existing-value)))))
-   :pre-create-post-validate
-   (wrap-auth :create (fn [auth-value doc-field payload]
-                        (update-in payload [doc-field]
-                                   (fn [existing-value]
-                                     (if (and existing-value (not= (str existing-value) (str auth-value)))
-                                       (throw (ex-info "Forbidden" {:type :forbidden}))
-                                       auth-value)))))
-   :pre-update-pre-validate
-   (wrap-auth :update (fn [auth-value doc-field payload]
-                        (update-in payload [doc-field]
-                                   (fn [existing-value]
-                                     (if-not existing-value
-                                       auth-value
-                                       existing-value)))))
-   :pre-update-post-validate
-   (wrap-auth :update (fn [auth-value doc-field payload]
-                        (update-in payload [doc-field]
-                                   (fn [existing-value]
-                                     (if (and existing-value (not= (str existing-value) (str auth-value)))
-                                       (throw (ex-info "Forbidden" {:type :forbidden}))
-                                       auth-value)))))
```

Comment:
Reparenting resources seems like a possible bug, so I removed that

Reply:
a bug ? It isn't enabled on partial update

Reply:
Oh right, I forgot that `update` hooks only mached `PUT` requests.